docs/research/tokenization.md

```diff
@@ -0,0 +1,301 @@
+
+### Maksymalna tokenizacja, [Byte Pair Encoding][bpe], różny rozmiar słowozbioru
+
+|Rozmiar słowozbioru (`vocab_size`)|Tokenizator             |Czas trwania [mm:ss]|Sum. l. tokenów|Śr. l. tokenów na utwór|Zmiana[{sup}`?`](#tip-zmiana) |
+|----------------------------------|------------------------|--------------------|---------------|-----------------------|-------|
+|                              500 |[REMI][remi]            |              03:14 |      19780309 |              15501.81 |-22.8% |
+|                             1000 |[REMI][remi]            |              04:05 |      17436890 |              13665.27 |-31.9% |
+|                             1500 |[REMI][remi]            |              04:16 |      16270492 |              12751.17 |-36.5% |
+|                             2000 |[REMI][remi]            |              04:43 |      15473773 |              12126.78 |-39.6% |
+|                             3000 |[REMI][remi]            |              05:45 |      14317688 |              11220.76 |-44.1% |
+
+:::{admonition} Uwaga
+:class: attention
+:name: tip-zmiana
+Dla danych per tokenizator, kolumna `Zmiana` podaje procentową zmianę sumarycznej liczby tokenów
+względem domyślnej tokenizacji
+bez BPE.
+:::
+
+:::{admonition} Uwaga
+:class: attention
+:name: tip-zmiana2
+Dla miar statystycznych, kolumna `Zmiana` podaje procentową zmianę sumarycznej liczby tokenów
+względem domyślnej tokenizacji
+bez BPE biorąc pod uwagę tylko te tokenizatory, które wspierają BPE.
+:::
+
+:::{admonition} Informacja
+:class: tip
+:name: tip-cut
+Średnia obcięta i odchylenie standardowe obcięte to odpowiednie miary policzone dla zbioru po
+odrzuceniu najwyższej i najniższej wartości.
+:::
+
+## Analiza wyników
+
+Czas pracy większości tokenizatorów dla domyślnej i maksymalnej konfiguracji był podobny i nie
+odstawał od średniej arytmetycznej o więcej, niż wartość odchylenia standardowego. W obu przypadkach
+najdłużej pracował [CPWord][cpword], o czasie pracy odstającym od średniej o ponad dwukrotność
+odchylenia standardowego. Prawdopodobnie wynika to z wykorzystywanej przez ten tokenizator
+zaawansowanej reprezentacji wyjściowej sekwencji.
+
+Dla konfiguracji domyślnej, sumaryczna liczba tokenów dla większości tokenizatorów jest bardzo
+bliska średniej. Wyjątkami są cztery tokenizatory: [Octuple][octuple] zwracał znacznie mniej
+tokenów, a [REMI+][remiplus] i [MuMIDI][mumidi] więcej.
+Różnica w liczbie tokenów pomiędzy najmniejszą i największą wartością wynosiła aż rząd
+wielkości (!).
+
+Zmiana konfiguracji nie zmieniła wyników dla tokenizatorów [Structured][structured] i [Octuple][octuple].
+Dla reszty konfiguracji czas pracy wzrósł średnio o 13% (najwięcej dla [CPWord][cpword], 20%), a
+liczba tokenów o 6%. Zmiana liczby tokenów jest jednak bardziej różnorodna: [CPWord][cpword] i
+[MMM][mmm] stworzyły mniej niż 1% więcej tokenów (przy wydłużonym czasie pracy), ale dla
```

Comment:
może "tylko" zamiast "mniej" bo zaraz potem masz "więcej" i się dziwnie to czyta